### PR TITLE
app-misc/figlet: fix building on musl

### PR DESCRIPTION
**Pull request by dm9pZCAq**
on `musl` there is no `sys/cdefs.h` so `__BEGIN_DECLS` and `__END_DECLS` are undefined

and as it's only needed for `c++`, we can remove it

`utf8.h`:
```c
__BEGIN_DECLS

size_t		utf8_to_wchar(const char *in, size_t insize, wchar_t *out,
		    size_t outsize, int flags);
size_t		wchar_to_utf8(const wchar_t *in, size_t insize, char *out,
		    size_t outsize, int flags);

__END_DECLS
```

Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: dm9pZCAq <v@0x0c.xyz>

### CODE REVIEW

File: app-misc/figlet/figlet-9999.ebuild
Comment (arkamar) [inline app-misc/figlet/figlet-9999.ebuild:16]: This is not needed, becuase this issue is already fixed in upstream.

File: app-misc/figlet/figlet-2.2.5-r1.ebuild
Comment (arkamar) [inline app-misc/figlet/figlet-2.2.5-r1.ebuild:17]: I would suggest to apply upstream patch https://github.com/cmatsuoka/figlet/commit/88428a728e1f99dcdc63d7c5ff0cd8b6cda37576 with `PATCHES=( ... )` (see [documentation](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html) for more information) rather than this.

**Review by arkamar — CHANGES_REQUESTED** [review]

File: app-misc/figlet/metadata.xml
Comment (juippis) [inline app-misc/figlet/metadata.xml:11]: You don't **have** to become a maintainer, you can just fix packages if you want.

**Review by juippis — COMMENTED** [review]
Your commit is missing the mandatory proper sign-off line. please see
https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin

**Comment by dm9pZCAq** [comment]
@juippis, why it's missing?

```sh
git show HEAD | sed 11q
```
```
commit a42785869321aef1fd89d9802e3775fde14dc8af
gpg: Signature made Thu Apr 15 17:56:09 2021 EEST
gpg:                using RSA key 31A0C087810468787F75E68236A3640CA194FB38
gpg: Good signature from "dm9pZCAq (for git) <v@0x0c.xyz>" [ultimate]
Author: dm9pZCAq <v@0x0c.xyz>
Date:   Thu Apr 15 17:56:09 2021 +0300

    app-misc/figlet: fix building on musl

    Package-Manager: Portage-3.0.18, Repoman-3.0.3
    Signed-off-by: dm9pZCAq <v@0x0c.xyz>

```

**Comment by juippis** [comment]
Please read the link I provided, to a correct section. It takes a minute.

**Comment by dm9pZCAq** [comment]
@juippis, that is, without "legal name", I can't commit to Gentoo?

**Comment by juippis** [comment]
Yes, it's a requirement for all commits in ::gentoo repository.

**Pull request closed without merging**
